### DISCUSSION TITLE
麻烦问下 windows 端使用tdengine 的dll 程序如何才能编译出来

### DISCUSSION THREAD

**Discussion by azhuoinfogs** [taosdata/TDengine#4632] [General]
**General Questions**
A general question should be related to TDengine's user's manual, user experience, milestones, business cooperations etc. 
Before you ask a question, please:
1. Try searching for answers in the official documentation
2. Try searching in the existing issues on GitHub
3. Try searching on tech forums such as Stack Overflow, Quora, CSDN, v2ex etc.
4. Try using online searching engines such as Google, Bing, Baidu

If all the ways above couldn't help, please clearly and concisely describe what the question is.

**Comment by sangshuduo** [comment]
README中有如下方法介绍：

On Windows platform
If you use the Visual Studio 2013, please open a command window by executing "cmd.exe". Please specify "x86_amd64" for 64 bits Windows or specify "x86" is for 32 bits Windows when you execute vcvarsall.bat.

mkdir debug && cd debug
"C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\vcvarsall.bat" < x86_amd64 | x86 >
cmake .. -G "NMake Makefiles"
nmake
If you use the Visual Studio 2019, please open a command window by executing "cmd.exe". Please specify "x64" for 64 bits Windows or specify "x86" is for 32 bits Windows when you execute vcvarsall.bat.

mkdir debug && cd debug
"c:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvarsall.bat" < x64 | x86 >
cmake .. -G "NMake Makefiles"
nmake
Or, you can open a command window by clicking Visual Studio 2019 menu "Tools -> Command Line -> Developer Command Prompt" or "Tools -> Command Line -> Developer PowerShell" then execute commands as follows:

mkdir debug && cd debug
cmake .. -G "NMake Makefiles"
nmake
